lib/shared/sse-connection/src/lib/SSEConnection.ts

```diff
@@ -0,0 +1,49 @@
+import type { DVCLogger } from '@devcycle/types'
+
+export class SSEConnection {
+    private connection?: EventSource
+
+    constructor(
+        private url: string,
+        private onMessage: (message: unknown) => void,
+        private logger: DVCLogger,
+    ) {
+        this.openConnection()
+    }
+
+    private openConnection() {
+        if (typeof EventSource === 'undefined') {
+            this.logger.warn(
+                'StreamingConnection not opened. EventSource is not available.',
+            )
+            return
+        }
+        this.connection = new EventSource(this.url, { withCredentials: true })
+        this.connection.onmessage = (event) => {
+            this.onMessage(event.data)
+        }
+        this.connection.onerror = () => {
+            this.logger.warn(
+                'StreamingConnection warning. Connection failed to establish.',
+            )
+        }
+        this.connection.onopen = () => {
+            this.logger.debug('StreamingConnection opened')
+        }
+    }
+
+    isConnected(): boolean {
+        return this.connection?.readyState === this.connection?.OPEN
```

Comment:
this is so much cleaner and easier than having a threadsafe version :p

Reply:
yea, this is just copying the JS SDK stuff into its own lib, still need to do some updates in a stacked PR for the NodeJS SDK.